posthog/hogql/functions/mapping.py

```diff
@@ -161,10 +161,9 @@ class HogQLFunctionMeta:
     "toFloat": HogQLFunctionMeta("toFloat64OrNull", 1, 1),
     "toDecimal": HogQLFunctionMeta("toDecimal64OrNull", 1, 1),
     "toDate": HogQLFunctionMeta(
-        "toDateOrNull",
```

Comment:
I'm unsure about this change. This may have unintended effects in existing HogQL queries, maybe?

Reply:
Forgot to submit the comment i had:

> Can I change this to use toDate without mapping? Not sure if this bug has always been around or if it's due to the version change but toDateOrNull isn't timezone aware but the printer is trying to parse in 'UTC' as the second argument

right now toDateOrNull doesn't work because of the second argument so I think making any change is fine?

Reply:
This is here for a good reason: Running `toDate()` on malformed input results in the whole query failing, which is a no-go, as this overload is used on data which is not at all guaranteed to be parsable as dates (otherwise we'd go to the stricter `(ast.DateTimeType, ast.DateType)` branch)

Reply:
Simply put, `toDate()` results in queries mysteriously failing for users with no recourse

Reply:
As for the the second argument not being supported, that's annoying 🤔 I think in that case we need to output this `toTimeZone(toDateOrNull(<1>), <2>)` – _seems_ to be equivalent to `toDate(<1>, <2>)`. But of course we're talking about timezones, so it may be in fact be wrong half the time

Reply:
toTimeZone(toDateOrNull(<1>), <2>) will error when toDateOrNull returns a null which seems counter to the usefulness of being null safe?

Reply:
Hmm, can you reproduce in Metabase for me? The rule of thumb with SQL functions is they quietly return NULL when the main arg is NULL, seems to hold for `toTimeZone()` for me too